doc/releases/release-notes-4.1.rst

```diff
@@ -97,6 +97,26 @@ Deprecated APIs and options
   1. when Stream Flash is not configured to do erase on its own
   2. when erase is used for removal of a data prior or after Stream Flash uses the designated area.
 
+* The pipe API has been reworked.
```

Comment:
tempted to suggest that this should only list the actually deprecated options and not the removed ones, reasoning: deprecated options are not going to break the application when upgrading to this release, removed one do, so deprecated ones are fine to have here but everything else should only live in the migration guide and maybe we could have a single link at the top referring to that

that's for @kartben and I, I think this is good for now